pkg/validation/projects.go

```diff
@@ -58,7 +58,11 @@ func ValidateProjects(projects []v1alpha2.Project) error {
 
 		if len(gitSource.Remotes) == 0 {
 			errString += fmt.Sprintf("\nprojects %s should have at least one remote", project.Name)
-		} else if len(gitSource.Remotes) > 1 || gitSource.CheckoutFrom.Remote != "" {
+		} else if len(gitSource.Remotes) > 1 || (gitSource.CheckoutFrom != nil && gitSource.CheckoutFrom.Remote != "") {
+			if len(gitSource.Remotes) > 1 && (gitSource.CheckoutFrom == nil || (gitSource.CheckoutFrom != nil && gitSource.CheckoutFrom.Remote == "")) {
+				errString += fmt.Sprintf("\nproject %s has more than one remote defined, but has no checkoutfrom remote defined", project.Name)
+				continue
+			}
```

Comment:
```suggestion
		} else if len(gitSource.Remotes) > 1 && gitSource.CheckoutFrom != nil {
			err := validateRemoteMap(gitSource.Remotes, gitSource.CheckoutFrom.Remote, project.Name)
			if err != nil {
				errString += fmt.Sprintf("\n%s", err.Error())
			}
		} else if len(gitSource.Remotes) > 1 && gitSource.CheckoutFrom == nil {
				errString += fmt.Sprintf("\nproject %s has more than one remote defined, but has no checkout from remote defined", project.Name)
		}
```

I would suggest breaking it to different else if conditions. My suggestion handles all the test cases you wrote but if we want to explicitly handle`""` checkoutfrom then we can do it the way you suggested,

Reply:
I believe this suggestion doesn't cover
```yaml
git:
  remotes: ["my-remote"]
  checkoutFrom:
    remote: "another-remote"
```
but I didn't read too closely.

:+1: on splitting this logic up though, it's very complicated.